### ISSUE TITLE
Compiled code size increased

### ISSUE THREAD

**Issue by vobruba-martin** [google/closure-compiler#2043]
My compiled code size is increased by ~5% (~10KB) after [this commit](https://github.com/google/closure-compiler/commit/82f23897059d244d793518a1355fe2e1cc0ba80f). Also typed percentage decreased from 99.95% to 94.64% (I'm using `--use_types_for_optimization`) and I get no warnings with `reportUnknownTypes`.

When I was trying to debug this issue I got this code:
```javascript
/**
 * @param {!MessagePort} port
 */
function connectionCreate(port) {};

/**
 * @param {!MessagePort} port
 */
function connectionPortRenew(port) {
	port.onmessage = null;
};
```

Before that commit this code is 100% typed and after it is 80% typed. BTW if you remove one of these functions (no matter which one) you'll get 100% typed which is weird.

I wasn't able to reproduce the compiled code increase but it seems to me that some type info from externs gets lost which leads to `--use_types_for_optimization` having no effect.

For @dimvar  :
> Can you do the following, to help me debug this?

I will try tomorrow.

> For the example you mention above, I don't know exactly how to use it yet. What is the definition of `MessagePort`? Did you just build this code without any other definitions, and the default externs, and got the typing percentage difference?

`MessagePort` is in [standard externs](https://github.com/google/closure-compiler/blob/master/externs/browser/html5.js#L1954). And yes, I built this code without any other definitions and got the typing percentage difference.

**Comment by vobruba-martin** [comment]
So I tried https://github.com/dimvar/closure-compiler/commit/067e4b233ec6a00a133ee1ccd1edc021b5bb85f0 and https://github.com/dimvar/closure-compiler/commit/2cbad239b68d4da75cacac5361636ecb45f9098d but both outputs are the same. Anything else I can try?

**Comment by vobruba-martin** [comment]
The issue disappears if I revert this change https://github.com/vobruba-martin/closure-compiler/commit/be905a6db8bd2e72b547d5a2008787a4c313d71f

**Comment by dimvar** [comment]
I'm a bit confused. You are saying that you don't see the size regression if you put the OTI run right before j2clChecksPass. Do you see the size regression if you put the OTI run right before gatherExternProperties? The pass j2clChecksPass should have no impact here. If it does, that's definitely a bug.

**Comment by dimvar** [comment]
Can you try one more thing? Place the OTI run right after j2clChecksPass, in the getChecks method, rather than right before gatherExternProperties, in the getOptimizations method. If that avoids the code size regression, then the culprit pass is `normalize`, which happens at the beginning of optimizations but is not visible in defaultpassconfig.

**Comment by vobruba-martin** [comment]
I tried to move `addOldTypeCheckerPasses` into `getChecks()` method and the issue is gone. There is no difference if it is after or before `j2clChecksPass`.

If I move it into `getOptimizations()` method the issue is back and there is no difference if it is after or before `gatherExternProperties` pass.

**Comment by dimvar** [comment]
Great, thanks for helping me debug this Martin. It's the normalize pass that causes the issue. I will move the conversion at the end of checks then.

**Comment by vobruba-martin** [comment]
You're welcome! Sadly I have no test case. It looks like a more complex code is needed for discovering the issue. Thanks for fixing this!

**Comment by dimvar** [comment]
I submitted the fix internally. It will land externally later today.

**Comment by Dominator008** [comment]
Fixed by 8649b292f5d59e133c86c6ba9f1c03db69da27f3.
